code/lspgen/src/lspgen_config.c

```diff
@@ -999,6 +1004,18 @@ lspgen_read_capability_config(lsdb_node_t *node, json_t *obj)
 	attr_template.key.attr_cp[2] = OSPF_TLV_SID_LABEL_RANGE;
     }
 
+    if (ctx->protocol_id == PROTO_ISIS) {
+        arr = json_object_get(obj, "sr_algo_list");
+        if (arr && json_is_array(arr)) {
+            num_arr = json_array_size(arr);
+	    attr_template.key.cap.sr_algo_len = num_arr;
+	    attr_template.key.cap.sr_algo = (uint8_t *)malloc((num_arr) * sizeof(uint8_t));
```

Comment:
I would use `calloc(num_arr, sizeof(uint8_t))` instead

Reply:
Ack, change incoming soon.

Reply:
use valgrind for checking against memleaks - e.g.
```
!/bin/sh

if ! which valgrind >/dev/null; then
    echo "`basename $0`: valgrind not found." >&2
    exit 1
fi

valgrind -v --tool=memcheck --trace-children=yes \
        --track-fds=yes --num-callers=32 \
        --memcheck:leak-check=yes --memcheck:leak-resolution=high \
        --show-reachable=yes \
        "$@" 2>&1 | tee vg.log
```

and

```./vg ./lspgen <argument list>```

Reply:
Thank you! Found it and freed now. Could you please review latest commit [added free for sr_algo](https://github.com/rtbrick/bngblaster/pull/277/commits/0f7334d91f34045a4684191f77ef3d83226623f7)

Reply:
looks good !